README.md

```diff
@@ -1,113 +1,283 @@
+
+Recall what you learned about the `var` and `let` keywords. If a constant is created using `let`, it can never be changed, right?
+
+```swift
+let x = 5  // x can never be assigned a new value!
+```
+
+A variable created using `var` can be changed, though:
+
+```swift
+var y = 10
+y = 15  // y can be changed to 15!
+```
+
+When working with objects, `let` and `var` behave nearly the same way. Once you assign an instance of a class to a `let` constant, that constant can never be assigned to another instance:
+
+```swift
+let square1 = Square(topLeftCorner: (0.0, 0.0), lengthOfSide: 1.0)
+square1 = Square(topLeftCorner: (10.0, 10.0), lengthOfSide: 10.0)
+// An error will occur on the second line.
+// square1 cannot be assigned to another object!
+```
+
+However, if you assign an instance of a class to a `var` variable, you _can_ reassign it to another instance:
+
+```swift
+var square2 = Square(topleftCorner: (1.0, 1.0), lengthOfSide: 9.0)
+square2 = Square(topLeftCorner: (2.0, 3.0), lengthOfSide: 3.5)
+// This is fine!
+```
+
+This is because you're not _changing_—or _mutating_—the instance of a class; you're just assigning `square2` to a new instance.
+
+The same applies to the `Square` class's properties: Because they were declared as constants, you cannot change them after an instance of the class has been created:
+
+```swift
+var square3 = Square(topLeftCorner: (10.0, 9.8), lengthOfSide: 3.2)
+square3.lengthOfSide = 3.4  // This causes an error!
+```
+
+This is why the `Square` class is called an _immutable_ class.
+
+On the other hand, if you declared the constants using the `var` keyword, like this:
+
+```swift
+class Square {
+    var topLeftCorner: (Double, Double)
+    var lengthOfSide: Double
+    var area: Double {
+        return lengthOfSide * lengthOfSide
     }
-    
-    init(name: String, canFly: Bool, magicPowers: [String]) {
-        self.name = name
-        self.canFly = canFly
-        self.magicPowers = magicPowers
+
+    init(topLeftCorner: (Double, Double), lengthOfSide: Double) {
+        self.topLeftCorner = topLeftCorner
+        self.lengthOfSide = lengthOfSide
     }
-    
 }
+```
+
+You could, in fact, change the properties from outside the class:
+
+```swift
+var square4 = Square(topLeftCorner: (8.0, 7.0), lengthOfSide: 15.0)
+square4.lengthOfSide = 9.7  // This is okay!
+```
+
+And since `area` is a computed property, it would even print the correct area after `lengthOfSide` is changed:
+
+```swift
+var square4 = Square(topLeftCorner: (8.0, 7.0), lengthOfSide: 15.0)
+print("Area is \(square4.area)")
+// Prints "Area is 225.0"
+square4.lengthOfSide = 9.7
+print("Area is \(square4.area)")
+// Prints "Area is 94.09"
+```
+
+One thing to note: If you declare a constant (using `let`), the properties of the object _cannot_ be changed, even if they are variables. For example, this code will produce an error, because `square5` is a constant, even though its properties are variables:
+
```

Comment:
This is incorrect. This is true for structs, but not classes.

It's what makes structs so appealing in Swift. You have true immutability when you declare a struct instance with the let keyword because all of its properties could NEVER be changed underneath you even if they have the var keyword. But with a class, it's different.

A class declared with the let keyword only insures that we never refer to another instance of that particular class, we're stuck with the one we have through initialization but we're free to change any properties declared with var on that instance (even though the instance was declared with the let keyword).

This is valid code:

```swift
class Link {
    
    var savedZelda: Bool = false
    
}

let brian = Link()

brian.savedZelda = true
// brian.savedZelda is true
```

If we were to swap out the keyword `class` for `struct` in my Link example, the compiler would tell us that we "Cannot assign to property: 'brian' is a 'let' constant" - but when it's a class, there would be no error.

That all being said - I will merge this in and make the change myself, so no need to worry :)

@mdippery